packages/roosterjs-content-model-core/lib/corePlugin/copyPaste/utils/createCopyRange.ts

```diff
@@ -0,0 +1,90 @@
+import { adjustSelectionForCopyCut } from './adjustSelectionForCopyCut';
+import { onCreateCopyEntityNode } from '../../../override/pasteCopyBlockEntityParser';
+import { preprocessTable } from './preprocessTable';
+import { pruneUnselectedModel } from './pruneUnselectedModel';
+import type {
+    ContentModelDocument,
+    DOMSelection,
+    IEditor,
+    OnNodeCreated,
+} from 'roosterjs-content-model-types';
+import {
+    contentModelToDom,
+    createModelToDomContext,
+    isElementOfType,
+    isNodeOfType,
+    iterateSelections,
+    wrap,
+} from 'roosterjs-content-model-dom';
+
+/**
+ * @internal
+ * Exported only for unit testing
+ */
+export const onNodeCreated: OnNodeCreated = (modelElement, node): void => {
+    if (isNodeOfType(node, 'ELEMENT_NODE') && isElementOfType(node, 'table')) {
+        wrap(node.ownerDocument, node, 'div');
+    }
+    if (isNodeOfType(node, 'ELEMENT_NODE') && !node.isContentEditable) {
+        node.removeAttribute('contenteditable');
+    }
+    onCreateCopyEntityNode(modelElement, node);
+};
+
+/**
+ * Create the range for the "beforeCutCopy" event
+ * @param editor The editor object
+ * @param pasteModel The model of the selection
+ * @param selection The editor selection
+ * @param tempDiv A temporary DIV element used for cut/copy content
+ * @returns
+ */
+export function createCopyRange(
+    editor: IEditor,
+    pasteModel: ContentModelDocument,
+    selection: DOMSelection,
+    tempDiv: HTMLDivElement
```

Comment:
Better also hide the code of creating tempDiv into this function, so caller does not need to know how to pass it in.

We need to make sure each parameter of a public API is easy to understand